core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -819,7 +831,8 @@ trait OptimisticTransactionImpl extends TransactionalWrite with SQLMetricsReport
 
       // Fail if files have been deleted that the txn read.
       val readFilePaths = readFiles.map(f => f.path -> f.partitionValues).toMap
-      val deleteReadOverlap = removedFiles.find(r => readFilePaths.contains(r.path))
+      val deleteReadOverlap = removedFiles.find(r =>
+        isDataChangeOrConflict(r) & readFilePaths.contains(r.path))
```

Comment:
Is this single `&` on purpose? (I've never used it before and am curious if `&&` weren't really meant).

Reply:
Wow, great catch! Totally by accident but luckily not logically incorrect, just slightly less efficient:  
https://stackoverflow.com/a/25067787/6877213

Will fix shortly.